hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/UserDetailsFormatter.java

```diff
@@ -0,0 +1,175 @@
+     *            the name size of each name part
+     * @return the formatted user name (max 2 * expectedNameLength characters)
+     *         cannot be <null>
+     */
+    public static String loadAndFormatUsername(final String username, final int expectedNameLength) {
+        final UserDetails userDetails = loadUserByUsername(username);
+        if (!(userDetails instanceof UserPrincipal)) {
+            return userDetails.getUsername();
+        }
+
+        final UserPrincipal userPrincipal = (UserPrincipal) userDetails;
+
+        final String trimmedFirstname = trimAndFormatDetail(userPrincipal.getFirstname(), expectedNameLength);
+        final String trimmedLastname = trimAndFormatDetail(userPrincipal.getLastname(), expectedNameLength);
+
+        if (StringUtils.isEmpty(trimmedFirstname) && StringUtils.isEmpty(trimmedLastname)) {
+            return StringUtils.substring(userPrincipal.getLoginname(), 0, 2 * expectedNameLength);
+        }
+        return trimmedFirstname + DETAIL_SEPERATOR + trimmedLastname;
+    }
+
+    /**
+     * Format the current tenant. The information is loaded by the current
+     * session information.
+     * 
+     * @return the formatted user name (max 8 characters) can be <null>
+     */
+    public static String formatCurrentTenant() {
+        final UserDetails userDetails = getCurrentUser();
+        if (!(userDetails instanceof UserPrincipal)) {
+            return null;
+        }
+
+        final UserPrincipal userPrincipal = (UserPrincipal) userDetails;
+        return trimAndFormatDetail(userPrincipal.getTenant(), 8);
+    }
+
+    private static UserDetails getCurrentUser() {
+        final SecurityContext context = (SecurityContext) VaadinService.getCurrentRequest().getWrappedSession()
+                .getAttribute(HttpSessionSecurityContextRepository.SPRING_SECURITY_CONTEXT_KEY);
+        return (UserDetails) context.getAuthentication().getPrincipal();
+    }
+
+    private static String trimAndFormatDetail(final String formatString, final int expectedDetailLength) {
+        final String detail = StringUtils.defaultIfEmpty(formatString, StringUtils.EMPTY);
+        final String trimmedDetail = StringUtils.substring(detail, 0, expectedDetailLength);
+        if (StringUtils.length(detail) > expectedDetailLength) {
+            return trimmedDetail + TRIM_APPENDIX;
+        }
+        return trimmedDetail;
+    }
+
+    private static UserDetails loadUserByUsername(final String username) {
+        final UserDetailsService userDetailsService = SpringContextHelper.getBean(UserDetailsService.class);
+        try {
+            final UserDetails loadUserByUsername = userDetailsService.loadUserByUsername(username);
+            if (loadUserByUsername == null) {
+                throw new UsernameNotFoundException("User not found " + username);
+            }
+            return loadUserByUsername;
+        } catch (final UsernameNotFoundException e) { // NOSONAR
```

Comment:
At first glance empty catch but creating a dummy user below is not good for readability. What about this:

```
UserDetails  loadUserByUsername = null;

try {
    final UserDetailsService userDetailsService = SpringContextHelper.getBean(UserDetailsService.class);
    loadUserByUsername = userDetailsService.loadUserByUsername(username);
    if (loadUserByUsername == null) {
            throw new UsernameNotFoundException("User not found " + username);
    }
   
} catch (final UsernameNotFoundException e) {
     loadUserByUsername = new User(username, "", Collections.emptyList());
}

return loadUserByUsername;
```